app/controllers/claims_controller.rb

```diff
@@ -79,9 +79,19 @@ def check_page_is_in_sequence
 
     raise ActionController::RoutingError.new("Not Found for #{params[:slug]}") unless page_sequence.in_sequence?(params[:slug])
 
+    handle_magic_link if page_sequence.magic_link?
     redirect_to claim_path(current_journey_routing_name, next_required_slug) unless page_sequence.has_completed_journey_until?(params[:slug])
   end
 
+  def handle_magic_link
+    otp = OneTimePassword::Validator.new(params[:code], answers.sent_one_time_password_at)
+    if otp.valid? # && EligibleEyProvider.eligible_email?(journey_session.answers.email_address) # TODO eligible check once model merged: https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/3050
```

Comment:
is it ok to handle the eligibility check like this? 
we only have magic links for this scenario currently - can be moved to something more generic if we have a magic link for another scenario.